processor/logcountprocessor/processor.go

```diff
@@ -0,0 +1,177 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logcountprocessor
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/observiq/observiq-otel-collector/receiver/routereceiver"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+// processor is a processor that counts logs.
+type processor struct {
+	config    *Config
+	matchExpr *Expression
+	attrExprs map[string]*Expression
+	counter   *LogCounter
+	consumer  consumer.Logs
+	logger    *zap.Logger
+	cancel    context.CancelFunc
+	mux       sync.Mutex
+}
+
+// newProcessor returns a new processor.
+func newProcessor(config *Config, consumer consumer.Logs, matchExpr *Expression, attrExprs map[string]*Expression, logger *zap.Logger) *processor {
+	return &processor{
+		config:    config,
+		matchExpr: matchExpr,
+		attrExprs: attrExprs,
+		counter:   NewLogCounter(),
+		consumer:  consumer,
+		logger:    logger,
+	}
+}
+
+// Start starts the processor.
+func (p *processor) Start(ctx context.Context, _ component.Host) error {
+	processorCtx, cancel := context.WithCancel(ctx)
```

Comment:
@cpheps for reference. We originally debated on this, because previously, it was written the other way. I didn't think we could trust the passed in context on the chance that it was cancelled after everything was started.

Reply:
@cpheps any concerns with switching the behavior back?

Reply:
I went ahead and switched the behavior back to defining a background context.